src/components/SettingsContainer.tsx

```diff
@@ -32,15 +35,29 @@ export function SettingsItem(inputProps: PropsWithChildren<Partial<SettingsItemP
     props.children = <div>{props.children}</div>;
   }
 
-  const formattedLabel = props.labelFormatter ? props.labelFormatter(props.label) : props.label;
+  useEffect(() => {
+    if (props.label && typeof props.label === "string" && !props.htmlFor) {
+      console.warn(
+        "SettingsItem: Please set the 'htmlFor' attribute to support screen readers in setting '" + props.label + "'."
+      );
+    }
+  }, [props.label, props.htmlFor]);
+
+  let labelElement = <div></div>;
+  if (props.label) {
+    const formattedLabel = props.labelFormatter ? props.labelFormatter(props.label) : props.label;
```

Comment:
I don't think this is used anywhere, honestly. I was debating whether to remove it in this PR